### PR TITLE
[SKIP CI] zephyr/docker-build.sh: remove apt update / install libssl-dev

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits. Main one:

zephyr/docker-build.sh: remove apt update / install libssl-dev

The Zephyr image has been updated in
zephyrproject-rtos/docker-image#58

This will save some time and even better avoid llvm.org
glitches like this one:

https://github.com/thesofproject/sof/pull/4089/checks?check_run_id=2554961695

### CODE REVIEW

**Comment by marc-hb** [comment]
In https://github.com/thesofproject/sof/pull/4182/checks?check_run_id=2569292327 checkpatch complains that the URL in a commit message is too long. No other checkpatch warning.

This is a pure .github actions change so any other, non-github error can be ignored.

File: .github/workflows/zephyr.yml
Comment (fredoh9) [inline .github/workflows/zephyr.yml:23]: If we set the version, we need to update occasionally if docker image is updated.
It looks like zephyr's latest image is not stable. Why is so? Can they keep different tag until it get stabilized?

Comment (marc-hb) [reply .github/workflows/zephyr.yml:23]: Very good question. I've asked the image maintainers on the #infrastructure channel athttps://zephyrproject.slack.com/archives/CGNR3LCHM about a more `:stable` version but there is none. 

Quoting the same channel:
 > right now we might have a few updates as we are catching up after we merged the 2 dockers, but usually it should be less frequent

I believe we will NOT have to update for every single image release. For now this is just a build test so if it builds with a non-latest image then it is good enough and if it breaks then we update the image (hence the Zephyr SDK) "on-demand". We don't want to force developers to constantly update their toolchain either after all.

We already build Zephyr with the random, latest Zephyr commit of the day so maybe it's good to have one less moving part, should help point the finger when something outside of SOF breaks this.

**Review by fredoh9 — APPROVED** [review]
Looks good, thank you for the clarification

**Review by iuliana-prodan — APPROVED** [review]

**Review by aiChaoSONG — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
